Tensile/Source/lib/source/hip/HipSolutionAdapter.cpp

```diff
@@ -346,6 +346,16 @@ namespace Tensile
                                        &argsSize,
                                        HIP_LAUNCH_PARAM_END};
 
+            if(m_debug)
+            {
+                int numBlocks = 0;
+                int blockSize
+                    = kernel.workGroupSize.x * kernel.workGroupSize.y * kernel.workGroupSize.z;
+                HIP_CHECK_RETURN(hipModuleOccupancyMaxActiveBlocksPerMultiprocessor(
```

Comment:
I would like to understand this part a bit more.
- how the output look like? Could you share some example?
- Why 4th parameter(dynSharedMemPerBlk ) is 0? I do not quite understand the interface of this function.

Reply:
This prints the occupancy of the kernel, I was using it to check occupancy of stream-k vs non-stream-k kernel variants. Output looks like:

Occupancy = 2

dynSharedMemPerBlk is amount of dynamic LDS we plan to allocate. We don't use any dynamic LDS, we know how much shared memory is required when we generate the kernel, so this is set to 0. The function can check .amdhsa_group_segment_fixed_size or .group_segment_fixed_size in kernel header for how much fixed shared memory is required to do the occupancy calculation.

Reply:
Thanks for your explanation. I got it.